src/osd/osd_types.h

```diff
@@ -6655,4 +6655,19 @@ using missing_map_t = std::map<hobject_t,
   std::pair<std::optional<uint32_t>,
     std::optional<uint32_t>>>;
 
+/**
+ * op_queue_type_t
+ *
+ * Supported op queue types
+ */
+enum class op_queue_type_t : uint8_t {
+  WeightedPriorityQueue = 0,
+  mClockScheduler,
+  PrioritizedQueue,
+  mClockPriorityQueue,
+  unknown
```

Comment:
There are 3 queue types, so there should be exactly 3 enums (and I think they should match the class names for simplicity): WeightedPriorityQueue, PrioritizedQueue, mClockPriorityQueue.

unknown is not a queue type.  It's only possible use would be to represent an invalidly set config for later handling, but make_scheduler will simply crash the OSD if it sees anything other than WeightedPriorityQueue or mClockScheduler.

Remove unknown.  Modify get_op_queue_type in OSD::OSD() to print a useful error message (including the config that was set!) and then assert if it sees an invalid option.

mClockPriorityQueue and mClockScheduler seem to be the same thing.  Use mClockPriorityQueue to match the type name and remove mClockScheduler (unless I'm missing an actual distinction?).

The user visible printed type name should match the user configs -- wpq and mclock_scheduler.

Reply:
Sounds good.

Reply:
- Implemented the suggested changes
- In addition, I modified the outdated description for osd_op_queue option in osd.yaml.in